source/develop/release-management/features/gluster/gdeploy-cockpit-integration.html.md

```diff
@@ -0,0 +1,140 @@
+
+### Benefit to oVirt
+
+This feature will greatly improve the user experience of deploying oVirt and Gluster hyperconverged solution using oVirt Node NG. oVirt users will be benefited in following ways.
+
+* Easy deployment using Cockpit UI
+* Well integrated deployment of HCI solution
+* Consistent Gluster deployment using gdeploy
+
+### Dependencies
+gdeploy package should be installed and running. gdeploy requires password less ssh from the gdeploy host to other hosts where we want to deploy gluster. This needs to be done manually before starting deployment flow.
+
+**TBD:** explore a way to configure password less ssh using Cockpit itself.
+
+## Detailed Description
+
+ Existing Cockpit-Plugin can be extended to support this feature and following will be the user flow for deploying a HCI environment using Cockpit.
+
+### UI Flow
+
+ HCI Deployment flow can be triggered by the option 'HCI Deployment with Gluster' in HE section of Ovirt plugin. This will open the following wizard which
+ help to provide all the required information about the gluster deployment.
+
+
+  ![Deploy-With-Glusterfs](Deploy-With-Gluster.png)
+
+
+ This wizard will have following five steps.
+
+ * Hosts:
+   This will help to capture all host information like IP address in the gluster cluster.
+
+
+  ![gdeploy-Hosts](gdeploy-hosts.png)
+
+
+* Packages:
+  This is used for capturing the packages which needs to be installed. This can also take care of additional repos which needs to enabled to install those packages.
+  In case of RHEL systems, this can also takes care of registering the system with CDN and enabling the required repos.
+
+  ![gdeploy-Packages](gdeploy-packages.png)
+
+* Volumes:
+  This step is used to capture the all the gluster volume details which needs to be created by gdeploy
+
+
+  ![gdeploy-Volumes](gdeploy-volumes.png)
+
+
+* Bricks:
+  This step will be used to define the bricks for the glusterfs Volumes defined in previous step.
+
+
+  ![gdeploy-bricks](gdeploy-bricks.png)
+
+
+* Summary:
+  This will be the final step in the gdeploy flow and it will show a summary of all the inputs captured in the previous steps.
+  This will generate a gdeploy config file with all the inputs captured in previous step. This step will also create an answer file for Hosted-Engine deployment with default answers for HCI deployment and gluster volume information for hosted-engine storage domain.
+  User can review the answer files and click 'Finish' to start executing the gluster deployment using gdeploy. Progress of gdeploy execution will be shown as it executes different steps in the deployment.
+
+
+    ![gdeploy-summary](gdeploy-summary.png)
+
+
+    ![gdeploy-log](gdeploy-log.png)
+
+
+* Hosted-Engine Setup:
+  After completing the gluster deployment, it will start the standard Hosted-Engine deployment by invoking the 'hosted-engine --deploy' with the generated answer file. Standard HE deployment work flow will start from this step.
```

Comment:
Just to be clear, dashboard/src/helpers/HostedEngineSetup will need a change in order to accept some optional argument (for the file location).

After this, setup needs to be invoked directly from react, which will call hosted-engine --deploy, otherwise the setup UI won't appear.